### ISSUE TITLE
Libcal: color contrast

### ISSUE THREAD

**Issue by rschwab** [UCSCLibrary/SiteImprove#56]
[Siteimprove link](https://my2.siteimprove.com/Accessibility/624596/77184211135/NextGen/Issue/1?conformance=&siteTargetIssueKinds=1,2&wcagVersion=21&pageSegments=&ruleName=sia-r69&ruleId=69&issueKind=1&tagFilterType=2&siteTarget.wcagVersion=21&siteTarget.conformanceLevels=0,1,3,4&siteTarget.issueKinds=1,2)

- [x] Mini-calendar widget
![Image](https://github.com/user-attachments/assets/2398e05e-7d13-436c-83ba-e5fc406c35c0)

- [x] In person badge

![Image](https://github.com/user-attachments/assets/d909482f-df4d-4689-9ebb-debae4842cdc)

- [x] 3d printer badge

![Image](https://github.com/user-attachments/assets/9aeeed40-7f0e-4b31-bbaf-bc347de11cc0)

- [ ] Appointment schedule form datepicker

![Image](https://github.com/user-attachments/assets/f72a473a-8dcc-4022-b5d6-9f4a6c835c32)

- [x] Events mini calendar widget

![Image](https://github.com/user-attachments/assets/193f5ff7-888b-44ad-a724-bf9d56a6307e)

**Comment by JessWaggs** [comment]
@rschwab We could make the background of the mini calendar white and that would solve the contrast issue. Do you think that would look okay?
<img width="334" height="470" alt="Image" src="https://github.com/user-attachments/assets/aecd11fd-6693-4678-8bca-c8589975053b" />
Or, alternatively, changing the gray color of the date numbers from <#777> to <#666> will meet WCAG AA (but not AAA)

**Comment by JessWaggs** [comment]
Submitted an accessibility report to Springshare for the date picker on 7/21/25

**Comment by JessWaggs** [comment]
Asked Digital Scholarship to check contrast on their DSI event categories 7/21/25

**Comment by rschwab** [comment]
I like the white background for the mini-calendar. But I also feel like it wants a border. What do you think?

<img width="1920" height="540" alt="Image" src="https://github.com/user-attachments/assets/d4180dcd-c6ea-46ce-9463-8791cb9ba49e" />

**Comment by JessWaggs** [comment]
Yeah, the white with a border looks great! Let's do it!

**Comment by rschwab** [comment]
Done! 

For the events mini-calendar widget, I tried updating our custom css to see if we can change the widget styles, but we cannot. I've opened a ticket with Springshare for that one just now.

**Comment by rschwab** [comment]
Springshare answered this a while back and it languished in my inbox - sorry! The CSS can be edited in the [calendar look & feel settings](https://calendar.library.ucsc.edu/admin/calendar/546/look-and-feel). I've changed the text from #7F7F7F to #4A4A4A which now passes both AA and AAA. I've checked off the events mini-calendar widget in the ticket description. One more down!

**Comment by rschwab** [comment]
This issue still has many instances in SI reports today. I'm going to leave this one in-progress until I can review the information here to see if its still relevant.

**Comment by rschwab** [comment]
Fixed all the instances (except one, see below) found today, 4/29/26. Fixed:
- Red status text on [certain spaces](https://calendar.library.ucsc.edu/space/139576) with malfunctioning equipment
- Orange background of days with events [on mini-calendar](https://calendar.library.ucsc.edu/widget/events/minicalendar?iid=181&cal_id=546&audience=&c=&z=&tags=&l=5&target=_blank&sp=0&date=2026-05-01)
- Fixed [audience tags and 3d-printer tag](https://calendar.library.ucsc.edu/calendar/DSI_Training?cid=17302&t=g&d=0000-00-00&cal=17302&audience=2584&ct=61641&inc=0) colors

What remains is the main calendar, where the [event background color](https://calendar.library.ucsc.edu/) has low contrast with the black text. The event category colors can be changed in the [calendar configuration](https://calendar.library.ucsc.edu/admin/calendar/546/category#s-lc-tab-category). @JessWaggs do you want to pick a color for these or should I just lighten the blue until I get enough contrast?

**Comment by JessWaggs** [comment]
@rschwab So this background color thing is kind of a mess... This background color appears in two places: 1) on the "full" calendar page, which uses black text against the background, and 2) as a badge on the event page, which uses white link text. So we would need a background color that works for both black and white text, which I'm not sure exists! 

I think the answer might be to override the styling for the main calendar by using the calendar level CSS. I was able to style the events to be our Design System light blue. However, my skills aren't up to styling the box that only appears on hover. Could you style that part to make it also #F3FAFF?

<img width="1539" height="551" alt="Image" src="https://github.com/user-attachments/assets/a3f923bf-743b-482f-88f5-e26f87b1425f" />

**Comment by rschwab** [comment]
Ok I think this one is now resolved! I was able to get the background color to stick on the popup using the !important keyword.

<img width="1000" height="566" alt="Image" src="https://github.com/user-attachments/assets/86863972-a665-4d4f-b7c9-9bf6f8019758" />
